JS SDK Revamp (#725)
rudderlabs/rudderstack-docs: docs/sources/sdks/rudderstack-javascript-sdk/filtering.mdx

---
title: "Filtering"

> description: Detailed guide on different types of filtering operations supported by the JavaScript SDK.

Reviewer: ```suggestion
description: Detailed guide on different types of event and destination filtering operations supported by the JavaScript SDK.
```

> ---

Reviewer: ```suggestion
aliases: ["event filtering", "destination filtering", "js sdk event filtering", "JavaScript SDK destination filtering"]
---
```

This guide covers how to achieve the following filtering operations:
* Filtering events while sending them to the specified destinations.

> * Filtering destinations where the event data should be sent.

Reviewer: ```suggestion
* Filtering destinations where the SDK sends the event data.
```

## Filtering events

> When sending events to a destination via the **web device mode**, you can explicitly specify which events should be discarded or allowed to flow through - by allowlisting or denylisting them using the <a href="https://rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/event-filtering/">Client-side Event Filtering</a> feature.

Reviewer: ```suggestion
When sending events to a destination via the <Link to="/destinations/rudderstack-connection-modes/#device-mode">**web device mode**</a>, you can specify which events should be discarded or allowed to flow through - by allowlisting or denylisting them using the <a href="https://rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/event-filtering/">Client-side Event Filtering</a> feature.
```

## Filtering destinations

You can send the event data only to the selective destinations by passing an <Link to="/sources/sdks/rudderstack-javascript-sdk/load-js-sdk/#integrationopts">integrations</Link>  object in the **loadOptions** parameter. RudderStack then loads or sends events only to the specified and enabled destinations.

* A sample snippet to send event data only to the **Google Analytics** and **Intercom** destinations:

```javascript
rudderanalytics.load(WRITE_KEY, DATA_PLANE_URL, {
    integrations: {
        All: false,
        "Google Analytics": true,
        Intercom: true
    }
});
```

* A sample `track` method to send event data only to the **Google Analytics** destination:

```javascript
rudderanalytics.track(
    "Order Completed", {
        revenue: 30,
        currency: "USD",
        user_actual_id: 12345
    }, {
        integrations: {
            All: false,
            "Google Analytics": true
        }
    }
);
```

> The destinaton name inside the `integrations` object should be in the specified format, such as:

Reviewer: ```suggestion
The destination name inside the `integrations` object should be in the specified format, such as:
```

Reviewer: Can we add the link to all the supported integration names?
